pkg/registry/generic/rest/proxy.go

```diff
@@ -202,7 +206,40 @@ func (h *UpgradeAwareProxyHandler) dialURL() (net.Conn, error) {
 
 		return tlsConn, nil
 	default:
-		return nil, fmt.Errorf("Unknown scheme: %s", h.Location.Scheme)
+		return nil, fmt.Errorf("unknown scheme: %s", h.Location.Scheme)
+	}
+}
+
+func dialURLWithTransport(httpTransport *http.Transport, location *url.URL) (net.Conn, error) {
```

Comment:
Looked at combining again and it's not terrible, but I still think ease of reading is worth the extra lines.